### PR TITLE
Customized module from gitlab

### PR DESCRIPTION
**Pull request by uuleon**

### CODE REVIEW

File: sql/expression_resolver.go [hunk 1]
```diff
@@ -47,6 +47,15 @@ type engineSpec struct {
 	queue   string
 }
 
+type gitLabModule struct {
```
Comment (typhoonzero) [inline sql/expression_resolver.go:50]: `gitLabModel` ?

File: sql/expression_resolver.go [hunk 2]
```diff
@@ -47,6 +47,15 @@ type engineSpec struct {
...
+	ModuleName   string
+	ProjectName  string
+	Sha          string
+	PrivateToken string
+	SourceRoot   string
+	GitLabServer string
```
Comment (typhoonzero) [inline sql/expression_resolver.go:56]: Maybe define some URI schema could reduce the number of parameters, like `token@https://gitlab.yourhost.com/your/repo.git:sha:/path/to/dir:ModelClassName`

Comment (uuleon) [reply sql/expression_resolver.go:56]: Is this a standard URL format for `gitlab`?

File: sql/expression_resolver.go (hunk 1 above)
Comment (uuleon) [reply sql/expression_resolver.go:50]: Ok, will fix it

File: sql/expression_resolver.go (hunk 2 above)
Comment (typhoonzero) [reply sql/expression_resolver.go:56]: Not a common standard, note we also use `maxcompute://xxx` to identify the database drivers. This is a method to define a pattern and reduce the number of paramters.

**Review by typhoonzero — DISMISSED** [review]
LGTM, can merge when the CI of the test case passed.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**